Packs/AzureDevOps/Integrations/AzureDevOps/AzureDevOps.py

```diff
@@ -0,0 +1,1732 @@
+    limit = int(arg_to_number(args.get('limit') or '50'))
+
+    if page < 1 or limit < 1:
+        raise Exception('Page and limit arguments must be greater than 1.')
+
+    offset = (page - 1) * limit
+
+    response = client.pull_requests_list_request(project, repository, offset, limit)
+
+    readable_message = f'Pull Request List:\n Current page size: {limit}\n Showing page {page} out others that may exist.'
+
+    outputs = {"name": project, "Repository": {"name": repository}}
+    for pr in response.get('value'):
+        data = generate_pull_request_output(pr)
+
+        pr_data = data["Repository"].pop("PullRequest")
+
+        if not dict_safe_get(outputs, ["Repository", "id"]):
+            outputs = {"name": data["name"], "Repository": data["Repository"]}
+            outputs["Repository"]["PullRequest"] = []
+
+        outputs["Repository"]["PullRequest"].append(pr_data)
+
+    readable_output = generate_pull_request_readable_information(response.get('value'), is_multiple_response=True,
+                                                                 message=readable_message)
+
+    command_results = CommandResults(
+        readable_output=readable_output,
+        outputs_prefix='AzureDevOps.Project',
+        outputs_key_field='name',
+        outputs=outputs,
+        raw_response=response
+    )
+
+    return command_results
+
+
+def project_list_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """
+    Retrieve all projects in the organization that the authenticated user has access to.
+    Args:
+        client (Client): Azure DevOps API client.
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+
+    """
+    page = int(arg_to_number(args.get('page') or '1'))
+    limit = int(arg_to_number(args.get('limit') or '50'))
+
+    if page < 1 or limit < 1:
+        raise Exception('Page and limit arguments must be greater than 1.')
+
+    offset = (page - 1) * limit
+    response = client.project_list_request(offset, limit)
+    readable_message = f'Project List:\n Current page size: {limit}\n Showing page {page} out others that may exist.'
+
+    outputs = []
+    output_headers = ['name', 'id', 'state', 'revision', 'visibility', 'lastUpdateTime']
+
+    for project in response.get('value', []):
+        data = {key: project.get(key) for key in output_headers}
+        str_time = data.get('lastUpdateTime')
+        if str_time.endswith('Z'):
+            data['lastUpdateTime'] = FormatIso8601(datetime.strptime(str_time, DATE_FORMAT))
+
+        outputs.append(data)
+
+    readable_output = tableToMarkdown(
+        readable_message,
+        outputs,
+        headers=output_headers,
+        headerTransform=pascalToSpace
+    )
+
+    command_results = CommandResults(
+        readable_output=readable_output,
+        outputs_prefix='AzureDevOps.Project',
+        outputs_key_field='name',
```

Comment:
In order to stay coherent with the rest of the commands `outputs_key_field`

Reply:
I am not sure I understand the reasoning. ID is a unique key (as opposed to name for example) it makes little sense to use something else if we have the ID available. In terms of coherency, it is not much important since this is not seen by the customer.